frontend/integration-tests/tests/secrets.scenario.ts

```diff
@@ -289,3 +290,55 @@ xdescribe('Interacting with the create secret forms', () => {
     });
   });
 });
+
+describe('OCP-21248 - Add Secret to Workloads', () => {
+  const secretName = 'ocp21248-secret';
+  const resourceName = 'ocp21248-deploy';
+  const resourceKind = 'deployment';
+  const envPrefix = 'env-';
+  const mountPath = '/tmp/testdata';
+
+
+  beforeAll(async()=> {
+    // create deployment and secret
+    execSync(`kubectl run ${resourceName} --image=aosqe/hello-openshift -n ${testName}`);
+    execSync(`kubectl create secret generic ${secretName} --from-literal=key1=supersecret -n ${testName}`);
+  });
+
+  beforeEach(async() => secretsView.visitOneSecretPage(appHost, testName, secretName));
+
+  afterEach(() => {
+    checkLogs();
+    checkErrors();
+  });
+
+  describe('Add Secret to Workloads as Enviroment Variables', ()=> {
+    it('Add Secret to Deployment as Env', async() => {
+      await secretsView.addSecretToWorkloadAsEnv(resourceName, envPrefix);
+      await new Promise(resolve => (function checkForValues() {
+        const output = secretsView.getResourceJSON(resourceName, testName, resourceKind);
+        if ( JSON.parse(output).status.observedGeneration === 2 ) {
+          return resolve();
+        }
+        setTimeout(checkForValues, 2000);
+      })());
+      expect(secretsView.searchJSON('spec.template.spec.containers[0].envFrom[0].secretRef.name', secretName, resourceName, testName, resourceKind)).toBe(true);
+      expect(secretsView.searchJSON('spec.template.spec.containers[0].envFrom[0].prefix', envPrefix, resourceName, testName, resourceKind)).toBe(true);
+    });
+  });
+
+  describe('Add Secret to Workloads as Volume', ()=> {
+    it('Add Secret to Deployment as Vol', async() => {
+      await secretsView.addSecretToWorkloadAsVol(resourceName, mountPath);
+      await new Promise(resolve => (function checkForValues() {
```

Comment:
Tried to extract a function for this, but failed, didn't figure out where's the problem @jhadvig  Can you please help me with that? [my draft code](http://pastebin.test.redhat.com/754121)

Reply:
Let's skip the refactoring, it can be done afterwards. Lets make this working first :)

Reply:
Thanks